pkg/apis/batch/types.go

```diff
@@ -85,6 +85,19 @@ type JobTemplateSpec struct {
 	Spec JobSpec
 }
 
+// CompletionMode specifies how Pod completions of a Job are tracked.
+type CompletionMode string
+
+const (
+	// NonIndexedCompletion means that Pod completions of a Job are
+	// indistinguishable from each other.
+	NonIndexedCompletion CompletionMode = "NonIndexed"
+
+	// IndexedCompletion means that each Pod completion of a Job is tracked
+	// individually, being associated to a completion index.
```

Comment:
are there restrictions on what values `Parallelism` and `Completions` can take  when using this mode?

Reply:
Yes, but I'm keeping them in the field, as that's where people will take a look first.

Reply:
may be not in this case, but a field can exist in multiple places, and so we wouldn't repeat the description of each value on each field. The documentation is for the type, not really for the field.